src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -117,7 +124,60 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
 
             if (returnValue != null)
             {
-                this.context.SetOutput(returnValue);
+                if (orchestratorInfo.IsOutOfProc && returnValue is JObject)
+                {
+                    var jObj = returnValue as JObject;
+                    if (jObj["actions"] != null)
+                    {
+                        var commandSets = JsonConvert.DeserializeObject<AsyncCommand[][]>(jObj["actions"].ToString());
+
+                        foreach (AsyncCommand[] commandSet in commandSets)
+                        {
+                            var tasks = new List<Task>();
+                            foreach (AsyncCommand command in commandSet)
+                            {
+                                switch (command.ActionType)
+                                {
+                                    case AsyncActionType.CallActivity:
+                                        tasks.Add(this.context.CallActivityAsync(command.FunctionName, command.Input));
+                                        break;
+                                    case AsyncActionType.CreateTimer:
+                                        using (var cts = new CancellationTokenSource())
+                                        {
+                                            tasks.Add(this.context.CreateTimer(command.FireAt, cts.Token));
+                                            if (command.IsCanceled)
+                                            {
+                                                cts.Cancel();
```

Comment:
I'm not sure how timer cancellation was implemented or tested on the JavaScript side, but it looks to me that this code only handles the case where a timer was created and canceled in the same episode (I'm assuming the `commandSet` is all the commands issued within a particular episode) - i.e. we created a timer, immediately canceled it, and then yielded. Does this work correctly if I schedule a timer, schedule an activity, and then cancel the timer *after the activity completes*?

Reply:
I tested this scenario. It does.